cli/azd/pkg/azdo/pipeline.go

```diff
@@ -51,7 +51,7 @@ func pipelineExists(
 	client build.Client,
 	projectId *string,
 	pipelineName *string,
-) (bool, error) {
+) (*build.BuildDefinition, error) {
```

Comment:
With the change of return values, the name `pipelineExists` no longer makes sense since it infers a boolean response.  Consider updating to something like `findPipeline(...)` where you could expect a return value or `nil` when not found.